perturbation.f90

```diff
@@ -9861,79 +9823,63 @@ recursive subroutine symm_mat_element_vector_k(jrot,irow,ijterm,func,mat_t,no_di
     integer(ik) :: cnu_i(0:PT%Nclasses),cnu_j(0:PT%Nclasses)
     integer(ik) :: deg_i(0:PT%Nclasses),deg_j(0:PT%Nclasses)
     real(rk)    :: mat_elem
-    integer(ik) :: isize,jsize,ielem,jelem,k_i,k_j,tau_i,tau_j
+    integer(ik) :: isize,jsize,ielem,jelem, k_i,k_j,tau_i,tau_j
     integer(ik) :: jrow,ideg,jdeg,isym,jsym,iL,iR,iterm,jterm,icontr,jcontr
     real(rk)    :: vec_i(PT%max_deg_size),vec_j(PT%max_deg_size)
     !
     real(rk), dimension(:,:,:), allocatable :: hcontr
     !
-    !call TimerStart('Symmetrized Hamiltonian - one column')
-    !
     mat_t = 0 
     !
     cnu_i(:) = PT%contractive_space(:,irow)
     !
     isize = PT%Index_deg(irow)%size1
     !
-    ! AT: hcontr is now an array of irow * PT%max_deg_size^2. This way we can calculate all hcontr values in advance,
+    ! AT: hcontr is an array of irow * PT%max_deg_size^2. This way we can calculate all hcontr values in advance,
     ! collect them to root, then run the matelem calculation loop.
     ! The reduction to root is necessary as we are doing apparently random access over a distributed matrix.
     allocate(hcontr(PT%max_deg_size,PT%max_deg_size,irow))
+    hcontr = 0.0
     !
     do jrow = 1,irow
-       !
        if ( present(no_diagonalization).and.no_diagonalization.and.jrow/=irow ) cycle
-       !
+
        cnu_j(:) = PT%contractive_space(:,jrow)
-       !
        jsize = PT%Index_deg(jrow)%size1 
-       !
+
        do ideg = 1,isize
-          !
           deg_i(:) = PT%Index_deg(irow)%icoeffs(:,ideg)
-          !
           do jdeg = 1,jsize
-             !hcontr(ideg,jdeg) = 0.0_rk
-             !
              deg_j(:) = PT%Index_deg(jrow)%icoeffs(:,jdeg)
-             !
-             !iroot = contr(iclass)%iroot(cnu_i(iclass),deg_i(iclass))
-             !jroot = contr(iclass)%iroot(cnu_j(iclass),deg_j(iclass))
-             !
+
              icontr = PT%icase2icontr(irow,ideg)
              jcontr = PT%icase2icontr(jrow,jdeg)
-             k_i = PT%rot_index(cnu_i(0),deg_i(0))%k
-             k_j = PT%rot_index(cnu_j(0),deg_j(0))%k
-             tau_i = PT%rot_index(cnu_i(0),deg_i(0))%tau
-             tau_j = PT%rot_index(cnu_j(0),deg_j(0))%tau
-             !
-             ! Matrix elements 
-             !
+
              if (jcontr .lt. co_startdim .or. jcontr .gt. co_enddim) then
                hcontr(ideg,jdeg,jrow) = 0.0_rk
              else
-               hcontr(ideg,jdeg,jrow) = func(icontr,jcontr,jrot,k_i,k_j,tau_i,tau_j)
+               if (job%rotsym_do) then
```

Comment:
Haven't checked but I'm assuming `job` is in scope here. Even so, Is it worth having an explicit extra argument `do_rotsym` in the subroutine instead of relying on the global?

Reply:
Job seems to be a variable at module-scope so yes, it should be available here. I think it probably is worth having this as an explicit argument actually, yes.